spring-boot-starter/starter-client/spring/src/main/java/org/camunda/bpm/client/spring/impl/client/ClientRegistrar.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.client.spring.impl.client;
+
+import org.camunda.bpm.client.spring.annotation.EnableExternalTaskClient;
+import org.camunda.bpm.client.spring.impl.client.util.ClientLoggerUtil;
+import org.camunda.bpm.client.spring.impl.util.AnnotationUtil;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.context.annotation.ImportBeanDefinitionRegistrar;
+import org.springframework.core.type.AnnotationMetadata;
+
+public class ClientRegistrar implements ImportBeanDefinitionRegistrar {
+
+  protected static final ClientLoggerUtil LOG = ClientLoggerUtil.CLIENT_LOGGER;
+
+  protected static final String BEAN_NAME = "externalTaskClient";
+
+  protected Class<? extends ClientFactory> externalTaskClientFactoryClass;
+
+  public ClientRegistrar() {
+    this(ClientFactory.class);
+  }
+
+  public ClientRegistrar(Class<? extends ClientFactory> externalTaskClientFactoryClass) {
+    this.externalTaskClientFactoryClass = externalTaskClientFactoryClass;
+  }
+
+  @Override
+  public void registerBeanDefinitions(AnnotationMetadata importingClassMetadata,
+                                      BeanDefinitionRegistry beanDefinitionRegistry) {
+    registerExternalTaskClient(importingClassMetadata, beanDefinitionRegistry);
+  }
+
+  protected void registerExternalTaskClient(AnnotationMetadata importingClassMetadata,
+                                            BeanDefinitionRegistry beanDefinitionRegistry) {
+    if (beanDefinitionRegistry.containsBeanDefinition(BEAN_NAME)) {
```

Comment:
We should make this check independent of the bean name `externalTaskClient` since a bean with a different name exposed by the user should also skip the external task bean creation.

Reply:
I agree 👍

Reply:
Definitely 👍 😃

Reply:
👍